src/plugins/mer/mervirtualboxmanager.cpp

```diff
@@ -500,6 +518,91 @@ void MerVirtualBoxManager::setVideoMode(const QString &vmName, const QSize &size
     process->runAsynchronously(args);
 }
 
+bool MerVirtualBoxManager::setVdiCapacityMb(MerConnection *connection, int sizeMb)
+{
+    const QString &vmName = connection->virtualMachine();
+    qCDebug(Log::vms) << "VM:" << vmName << "Changing vdi size to" << sizeMb << "MB";
+
+    const VirtualMachineInfo virtualMachineInfo = fetchVirtualMachineInfo(vmName, true);
+    if (sizeMb < virtualMachineInfo.vdiCapacityMb)
+        return false;
+    else if (sizeMb == virtualMachineInfo.vdiCapacityMb)
+        return true;
+
+    QStringList arguments;
+    arguments.append(QLatin1String(MODIFYMEDIUM));
+    arguments.append(virtualMachineInfo.vdiPath);
+    arguments.append(QLatin1String(RESIZE));
+    arguments.append(QString::number(sizeMb));
+
+    QTime timer;
+    timer.start();
+
+    VBoxManageProcess process;
+    if (!process.runSynchronously(arguments)) {
+        qWarning() << "VBoxManage failed to" << MODIFYMEDIUM << virtualMachineInfo.vdiPath << RESIZE << sizeMb
+                   << "for VM" << vmName;
+        return false;
+    }
+
+    if (!increaseVmPatrition(connection)) {
```

Comment:
May be replace script like this to avoid restarting the device.
 And set autostart service on system start
```
#!/bin/bash

set -o nounset

resize() {
	# Compare available and current partition sizes
	pre=$(sfdisk -s /dev/sda1)
	post=$(sfdisk -s /dev/sda)

    if ! diff -q $pre $post >/dev/null 2>&1; then
    	echo "RESIZE:"
	    sfdisk --force -N 1 --no-reread /dev/sda <<<", +" || return
	    newsize=$(sfdisk --dump /dev/sda |sed -n 's,^/dev/sda1 : .*size=[[:space:]]*\([0-9]\+\).*,\1,p')
	    resizepart /dev/sda 1 "$newsize" || return
	    resize2fs /dev/sda1
    fi
}

resize || exit
```

Reply:
yup, that's my plan